clang/test/SemaCXX/attr-cxx0x.cpp

```diff
@@ -53,3 +53,14 @@ alignas(4) auto PR19252 = 0;
 
 // Check the diagnostic message
 class alignas(void) AlignasVoid {}; // expected-error {{invalid application of 'alignas' to an incomplete type 'void'}}
+
+namespace GH108819 {
+void a([[maybe_unused]] void) {}                 // expected-warning {{attribute 'maybe_unused' cannot be applied to a 'void' parameter}}\
```

Comment:
Can we add a test for a member function. also a templated function as well. Other combination if you can come up w/ them should also be tested. We need to do a better job of covering as much as possible.

Reply:
@shafik Thanks for the feedback. I've added more test cases. Could you take a look?

Reply:
Shafik's availability may be odd this week because of WG21 meetings in Hagenberg. The new test cases look good to me, so I think you're fine to land the PR and we can address any additional test coverage needs post-commit if Shafik has other cases in mind.